cdc/puller/ddl_puller.go

```diff
@@ -142,22 +193,201 @@ func (p *ddlJobPullerImpl) initJobTableMeta() error {
 		return errors.Trace(err)
 	}
 
-	p.tableInfo = model.WrapTableInfo(db.ID, db.Name.L, 0, tableInfo)
-	p.columnID = col.ID
+	p.ddlJobsTable = model.WrapTableInfo(db.ID, db.Name.L, 0, tableInfo)
+	p.jobMetaColumnID = col.ID
 	return nil
 }
 
 func (p *ddlJobPullerImpl) unmarshalDDL(rawKV *model.RawKVEntry) (*timodel.Job, error) {
 	if rawKV.OpType != model.OpTypePut {
 		return nil, nil
 	}
-	if p.tableInfo == nil && !entry.IsLegacyFormatJob(rawKV) {
+	if p.ddlJobsTable == nil && !entry.IsLegacyFormatJob(rawKV) {
 		err := p.initJobTableMeta()
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
 	}
-	return entry.ParseDDLJob(p.tableInfo, rawKV, p.columnID)
+	return entry.ParseDDLJob(p.ddlJobsTable, rawKV, p.jobMetaColumnID)
+}
+
+// handleRenameTables gets all the tables that are renamed
+// in the DDL job out and filter them one by one, if a table is
+// filtered, it will be remove from `job.BinlogInfo.MultipleTableInfos`.
+// If all the tables are filtered, skip will be true.
+func (p *ddlJobPullerImpl) handleRenameTables(job *timodel.Job) (skip bool, err error) {
+	var (
+		oldSchemaIDs, newSchemaIDs, oldTableIDs []int64
+		newTableNames, oldSchemaNames           []*timodel.CIStr
+	)
+
+	var (
+		remainOldSchemaIDs, remainNewSchemaIDs, remainOldTableIDs []int64
+		remainNewTableNames, remainOldSchemaNames                 []*timodel.CIStr
+	)
+
+	err = job.DecodeArgs(&oldSchemaIDs, &newSchemaIDs,
+		&newTableNames, &oldTableIDs, &oldSchemaNames)
+	if err != nil {
+		return true, errors.Trace(err)
+	}
+
+	multiTableInfos := job.BinlogInfo.MultipleTableInfos
```

Comment:
Seems that the `owner/schema.go` also has the similar logic, can we merge them?

Reply:
I have tried, but it is hard to do, these places split a `renameTables` job for different purposes.

Reply:
Maybe we can extract the basic logic in `handleRenameTables()` here and `schemaWrap4Owner.parseRenameTables` to a public place?